packages/react-core/src/components/Select/Select.tsx

```diff
@@ -470,6 +479,27 @@ class Select extends React.Component<SelectProps & InjectedOuiaProps, SelectStat
       );
     }
 
+    const generateSelectedBadge = () => {
+      let badgeText = null;
+      if (!isCheckboxSelectionBadgeHidden && selections && Array.isArray(selections)) {
+        if (enableZeroBadge && selections.length === 0) {
+          badgeText = 0;
+        }
+        if (selections.length > 0) {
+          badgeText = selections.length;
+        }
+        if (enableAllBadge && selections.length === maxNumSelections) {
+          badgeText = 'All';
```

Comment:
this text needs to set by prop fro localization.

Reply:
Sure, I'll refactor.

Reply:
@tlabaj - I pushed a refactored version; let me know if you have other feedback.